### DISCUSSION TITLE
Percentage of coverage of contigs (genomes)

### DISCUSSION THREAD

**Discussion by SergeyBaikal** [wwood/CoverM#147] [Q&A]
Dear developers! Thanks a lot for the program!
I have one question (or guesses)
I have bam files with genome aligned reads. To know the most covered genome, I need to know the percentage coverage for each genome. I did not find this function in the description. Is it advisable to do so? For example, I have several genomes, and by the percentage of coverage, I can judge the completeness of the coverage and thus identify the best coverage for the genome.
It seemed to me that this is done by the function relative_abundance 
But in this way it seems impossible to calculate the percentage of coverage for contigs.
`coverm genome --bam-files /home/sergey/CoverM/*.bam -m relative_abundance -t 16 --output-file /home/sergey/CoverM/out.txt
`

**Comment by wwood** [comment]
Hi,

Would covered_fraction suit?

-\-------------
Ben Woodcroft
Group leader, Centre for Microbiome Research, QUT
________________________________
From: Sergey Potapov ***@***.***>
Sent: Tuesday, December 27, 2022 10:40:20 AM
To: wwood/CoverM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [wwood/CoverM] Percentage of coverage of contigs (genomes) (Discussion #147)


Dear developers! Thanks a lot for the program!
I have one question (or guesses)
I have bam files with genome aligned reads. To know the most covered genome, I need to know the percentage coverage for each genome. I did not find this function in the description. Is it advisable to do so? For example, I have several genomes, and by the percentage of coverage, I can judge the completeness of the coverage and thus identify the best coverage for the genome.
It seemed to me that this is done by the function relative_abundance
But in this way it seems impossible to calculate the percentage of coverage for contigs.
coverm genome --bam-files /home/sergey/CoverM/*.bam -m relative_abundance -t 16 --output-file /home/sergey/CoverM/out.txt

―

**Reply by SergeyBaikal** [reply]
Thanks for the answer!
--min-covered-fraction default: 10   
10 Is it a percentage?

**Comment by wwood** [comment]
Yes

-\-------------
Ben Woodcroft
Group leader, Centre for Microbiome Research, QUT
________________________________

**Reply by SergeyBaikal** [reply]
Thank you!

**Comment by rhysnewell** [comment]
Just to clarify what Ben was saying. To get the covered fraction for input genomes you would want to use something like the following command:
```
coverm genome --bam-files /home/sergey/CoverM/*.bam -m relative_abundance covered_fraction trimmed_mean -t 16 --output-file /home/sergey/CoverM/out.txt 
```
That command will output both the relative abundance, fraction of the genome that is covered by reads, and the trimmed mean coverage. The `--min-covered-fraction` command controls which genomes/contigs to report as having any coverage at all. i.e. `--min-covered-fraction 10` will only report genomes which have >= 10% genome coverage.

Rhys

**Reply by SergeyBaikal** [reply]
Dear, Rhys! 
Fabulous! Thanks for the extended answer.
